docs/spec/v1/kustomizations.md

```diff
@@ -847,6 +847,15 @@ It has two fields:
 - `.secretRef.name`: The name of the secret that contains the keys to be used for
   decryption. This field can be omitted when using the
   [global decryption](#controller-global-decryption) option.
+- `.serviceAccountName`: The name of the service account used for
+  secret-less authentication with KMS services from cloud providers.
+  If a static credential for a given cloud provider is defined
+  inside the secret referenced by `.secretRef`, that static
+  credential takes priority.
```

Comment:
We want to allow both `secretRef` and `serviceAccountName` to be set because a user could be using a secret for a given cloud provider while using automatic/default authentication for another provider, all in the same Kustomization with SOPS-encrypted secrets using KMS keys from both cloud providers. They should still be able to do the same thing when using `serviceAccountName`, except that all the automatic authentication will now use the specified service account instead of the default one (the controller's).